### PR TITLE
refactor __main__ so that commands are not in a class

### PR DESCRIPTION
**Pull request by ctb**
This is solely deindentation and minor refactoring of `__main__` - no functionality change, but
the new structure permits more flexibility by moving the subcommands out of the class.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review and merge.

File: sourmash_lib/__main__.py [hunk 1]
```diff
@@ -41,883 +43,899 @@ def __init__(self):
...
         parser.add_argument('command')
         args = parser.parse_args(sys.argv[1:2])
-        if not hasattr(self, args.command):
+        if args.command not in self.commands:
             error('Unrecognized command')
             parser.print_help()
             sys.exit(1)
 
-        cmd = getattr(self, args.command)
+        cmd = globals().get(args.command)
```
Comment (luizirber) [inline sourmash_lib/__main__.py]: What about using a dict globally, and setting key to command as string and value as the function to be called?
```
commands = {
    'search': search,
    'watch': watch,
}
```

This allows to add commands from submodules later:
```
from sourmash_lib import submodule

commands = {
   'new_command': submodule.command
}
```
and other variations without needing to call `globals()` (and without risking having some other variable colliding with the command we want)

**Comment by ctb** [comment]
Sure - but I'd like to get this merged soon so that the massive whitespace
change doesn't interfere with other PRs :). Suits?

**Comment by luizirber** [comment]
Deal!

**Pull request merged**